src/dynlib/dynlib_windows.c

```diff
@@ -80,6 +124,14 @@ k4a_result_t dynlib_create(const char *name, uint32_t major_ver, uint32_t minor_
         }
     }
 
+    if (dllDirectory != 0)
```

Comment:
nit: NULL

Or switch `add_current_module_to_search` to return 0.

```suggestion
    if (dllDirectory != NULL)
```